src/scram.c

```diff
@@ -862,8 +862,21 @@ char *build_server_first_message(ScramState *scram_state, const char *username,
 				goto failed;
 			break;
 		case PASSWORD_TYPE_PLAINTEXT:
-			if (!build_adhoc_scram_secret(stored_secret, scram_state))
-				goto failed;
+			if (user->adhoc_scram_secrets_cached) {
```

Comment:
We should have some way to invalidate the cache when a new password is assigned to the user.

Reply:
Its a good point. Any thoughts on this? Maybe a TTL? Do we have any cache invalidation on PgBouncer code already?

Reply:
There is `unquote_add_authfile_user`, which is called for each user in the reloaded password file. That seems like a reasonable place to do this.

Reply:
Fixed